benches/shredder_benchmark.rs

```diff
@@ -0,0 +1,121 @@
+use std::cell::RefCell;
+
+use criterion::{black_box, Criterion, criterion_main};
+use criterion::criterion_group;
+use rand::rngs::StdRng;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+
+use shredder::{collect, number_of_tracked_allocations, Gc, Scan, run_with_gc_cleanup};
+
+// BENCHMARK 1: My janky stress test
+#[derive(Debug, Scan)]
+struct DirectedGraphNode {
+    label: String,
+    edges: Vec<Gc<RefCell<DirectedGraphNode>>>,
+}
+
+const NODE_COUNT: usize = 1 << 15;
+const EDGE_COUNT: usize = 1 << 15;
+const SHRINK_DIV: usize = 1 << 10;
+
+fn stress_test() -> Vec<usize> {
+    run_with_gc_cleanup(|| {
+        let mut nodes = Vec::new();
+
+        for i in 0..=NODE_COUNT {
+            nodes.push(Gc::new(RefCell::new(DirectedGraphNode {
+                label: format!("Node {}", i),
+                edges: Vec::new(),
+            })));
+        }
+
+        let mut rng = StdRng::seed_from_u64(0xCAFE);
+        for _ in 0..=EDGE_COUNT {
+            let a = nodes.choose(&mut rng).unwrap();
+            let b = nodes.choose(&mut rng).unwrap();
+
+            a.borrow_mut().edges.push(Gc::clone(b));
+        }
+
+        let mut res = Vec::new();
+        for i in 0..NODE_COUNT {
+            if i % SHRINK_DIV == 0 {
+                nodes.truncate(NODE_COUNT - i);
+                collect();
+                res.push(number_of_tracked_allocations());
+            }
+        }
+
+        res
+    })
+}
+
+pub fn benchmark_stress_test(c: &mut Criterion) {
+    c.bench_function("stress_test", |b| b.iter(|| black_box(stress_test())));
+}
+
+
+// BENCHMARK 2: It's binary-trees from the benchmarks game!
+
+fn count_binary_trees(max_size: usize) -> Vec<usize> {
```

Comment:
Do you ever actually remove anything with this test? Or are you looking to get signal about the overall cleanup?

Reply:
It's supposed to measure mostly allocation timings, automatically triggered collections, and then also the full sweep done by `run_with_gc_cleanup`